Graduate Job Pod Failure Policy to stable (#46807)
kubernetes/website: content/en/docs/concepts/workloads/controllers/job.md

If you do specify `.spec.parallelism` and `.spec.completions` both greater than 1, then there may be
multiple pods running at once. Therefore, your pods must also be tolerant of concurrency.

> When the `.spec.podFailurePolicy` field is set, the Job controller does not consider a terminating

Reviewer: ```suggestion
If you specify the `.spec.podFailurePolicy` field, the Job controller does not consider a terminating
```

nit: active voice

Author: Thanks, applied

> Pod (a pod that has a `.metadata.deletionTimestamp` field set) as a failure until that Pod is

Reviewer 2: I suppose `Once the job is finished, the Job status looks as follows:` should be also updated.

> Pod (a pod that has a `.metadata.deletionTimestamp` field set) as a failure until that Pod is

Author: > Maybe it would make sense to explain the FailureTarget under the Handling Pod and container failures.

Sure, I have expanded on this in the sibling PR: https://github.com/kubernetes/website/pull/46808

> Maybe as a new section as opposed to having it under Pod failure policy since its behavior is guarded by other 2 feature gates?

Yes, it is added in a new dedicated section in the sibling PR now.

> I suppose Once the job is finished, the Job status looks as follows: should be also updated.

Good catch, also updated there.

### Pod failure policy {#pod-failure-policy}

{{< feature-state for_k8s_version="v1.26" state="beta" >}}

{{< note >}}
You can only configure a Pod failure policy for a Job if you have the
`JobPodFailurePolicy` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
enabled in your cluster. Additionally, it is recommended
to enable the `PodDisruptionConditions` feature gate in order to be able to detect and handle
Pod disruption conditions in the Pod failure policy (see also:
[Pod disruption conditions](/docs/concepts/workloads/pods/disruptions#pod-disruption-conditions)).
Both feature gates are available in Kubernetes {{< skew currentVersion >}}.

> {{< /note >}}

Reviewer 3: I'm wondering if this note still would be valid since these feature gates still exist.
So, shouldn't we remove this note when we remove these feature gates?
But, I'm not sure when we should remove such a note. Is there any guideline for this situation?

Author: Yes, this is obsolete. Removed. PTAL

Author: > Is there any guideline for this situation?

I'm not sure about that, but it seems the practice, for example here: https://github.com/kubernetes/website/pull/46991/files, or here https://github.com/kubernetes/website/pull/45297/files

Reviewer 3: Oh, I see. Thank you for letting me know!

> {{< feature-state for_k8s_version="v1.31" state="stable" >}}

Reviewer 4: Improvement for 'feature-state' shortcode usage here, consider using the new feature_gate_name option, making the feature state more data-driven.
```suggestion
{{< feature-state feature_gate_name="JobPodFailurePolicy" >}}
```

Author: Thanks, done
